tests/unit/clients/test_ftp_client.py

```diff
@@ -87,6 +88,16 @@ def test_scandir_file(self, mocked_ftp_conn):
         assert entries[0].url == URL("ftp://localhost:9999/mydir/my_file.txt")
         assert entries[0].is_file
 
+    @pytest.mark.parametrize("fake_file_names", [['SA0006923', 'CR00095552']])
+    def test_scandir_ftp_dir(self, mocked_ftp_conn, fake_file_names):
+        client = ftp_client.FTPClient("ftp://localhost:9999/mydir")
+        with client:
+            client.conn.mlsd.side_effect = ftplib.error_perm
+            client.conn.dir.return_value = fake_file_names
```

Comment:
I don't know how to write this test to be honest. client.conn.dir uses a callback to append lines written out to a list. So it returns None.